src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py

```diff
@@ -403,83 +498,72 @@ def find_fixed_range(testcase_id, job_type):
+    # a bad state which we didn't catch through our usual means.
+    testcase = data_handler.get_testcase_by_id(output.uworker_input.testcase_id)
+    if data_handler.is_first_retry_for_task(testcase, reset_after_retry=True):
+      tasks.add_task('progression', output.uworker_input.testcase_id, output.uworker_input.job_type)
+      data_handler.update_progression_completion_metadata(testcase, output.testcase.crash_revision)
+      return
 
-def utask_main(uworker_input):
-  """Execute progression task."""
-  try:
-    find_fixed_range(uworker_input.testcase_id, uworker_input.job_type)
-  except errors.BuildSetupError as error:
-    # If we failed to setup a build, it is likely a bot error. We can retry
-    # the task in this case.
-    testcase = data_handler.get_testcase_by_id(uworker_input.testcase_id)
-    error_message = 'Build setup failed r%d' % error.revision
-    data_handler.update_testcase_comment(testcase, data_types.TaskState.ERROR,
-                                         error_message)
-    build_fail_wait = environment.get_value('FAIL_WAIT')
-    tasks.add_task(
-        'progression',
-        uworker_input.testcase_id,
-        uworker_input.job_type,
-        wait_time=build_fail_wait)
-  except errors.BadBuildError:
-    # Though bad builds when narrowing the range are recoverable, certain builds
-    # being marked as bad may be unrecoverable. Recoverable ones should not
-    # reach this point.
-    testcase = data_handler.get_testcase_by_id(uworker_input.testcase_id)
-    error_message = 'Unable to recover from bad build'
-    data_handler.update_testcase_comment(testcase, data_types.TaskState.ERROR,
-                                         error_message)
-  except errors.BuildNotFoundError as e:
-    # If an expected build no longer exists, we can't continue.
-    testcase = data_handler.get_testcase_by_id(uworker_input.testcase_id)
-    testcase.fixed = 'NA'
+    # The bug is fixed.
+    testcase.fixed = 'Yes'
     testcase.open = False
-    error_message = f'Build {e.revision} not longer exists'
-    data_handler.clear_progression_pending(testcase)
-    data_handler.update_testcase_comment(testcase, data_types.TaskState.ERROR,
-                                         error_message)
+    data_handler.update_progression_completion_metadata(
+        testcase, output.testcase.crash_revision, message='fixed on latest custom build')
+    return
 
+  # TODO: This should probably be moved to the end of the newly created progression_bisection task
   # If there is a fine grained bisection service available, request it. Both
   # regression and fixed ranges are requested once. Regression is also requested
   # here as the bisection service may require details that are not yet available
   # (e.g. issue ID) at the time regress_task completes.
-  testcase = data_handler.get_testcase_by_id(uworker_input.testcase_id)
+  testcase = data_handler.get_testcase_by_id(output.testcase.key.id())
```

Comment:
lets use output.uworker_input.testcase_id for this.
We don't want to trust the key here as it could allow the untrusted worker to mess with other testcases.

Reply:
Ping.

Reply:
I'm going to modify all instances to use `output.uworker_input.testcase_id`